### PR TITLE
[opt](meta) put normal partitions into recyclebin when replacing partitions

### LINKED ISSUE #36714: [Enhancement] (nereids) Support fore keyword in replace partition statment
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

When exec a sql like `alter table replace partition (p1) with temporary (tp1);`
The data of the normal partition `p1` will be dropped immediately.

### Solution

By default, move the date of normal partition to recycle bin.
Add the force keyword in the `replace partition` statement to indicate skipping the recycle bin and deleting directly.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liutang123**
## Proposed changes

Issue Number: close #36714

Move the old partitions to recycle bin when replacing partition.

### CODE REVIEW

**Comment by dataroaring** [comment]
Please handle truncate too, and add regression test.

**Comment by dataroaring** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

**Review by zddr — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/persist/ReplacePartitionOperationLog.java [hunk 1]
```diff
@@ -53,11 +54,13 @@ public class ReplacePartitionOperationLog implements Writable {
     private long version = 0L;
     @SerializedName(value = "versionTime")
     private long versionTime = 0L;
+    @SerializedName(value = "force")
+    private boolean force = FeConstants.DEFAULT_DROP_NORMAL_PARTITION_WHEN_REPLACE_PARTITION;
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/persist/ReplacePartitionOperationLog.java:58]: default to false.

**Comment by liutang123** [comment]
run buildall

**Review by zclllyybb — COMMENTED** [review]
It should be implemented in nereids first

**Comment by liutang123** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by liutang123** [comment]
rub buildall

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run p0

**Comment by liutang123** [comment]
run p0

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**